servers/src/mining/stratumserver.rs

```diff
@@ -574,16 +575,21 @@ impl Handler {
 					// set a new deadline for rebuilding with fresh transactions
 					deadline = Utc::now().timestamp() + config.attempt_time_per_block as i64;
 
-					self.workers.update_block_height(new_block.header.height);
-					self.workers
-						.update_network_difficulty(state.current_difficulty);
-
+					// Manage list of versions of this heights block
 					if clear_blocks {
 						state.current_block_versions.clear();
 					}
 					state.current_block_versions.push(new_block);
-					// Send this job to all connected workers
+
+					// Update the mining stats
+					self.workers
+						.update_block_height(state.current_block_versions[0].header.height);
```

Comment:
Why not keep `self.workers.update_block_height(new_block.header.height);`?

Reply:
Sure.  That new_block was just pushed to the current_block_versions[0] so its the same thing.  I'll change it back.

Reply:
Now I recall why - new_block has been "moved" when it gets pushed onto the block version list.   But it seems reasonable to set the stats first then push it onto the list.